NDTensors/src/lib/BlockSparseArrays/test/test_basics.jl

```diff
@@ -210,6 +204,15 @@ using .NDTensorsTestUtils: devices_list, is_supported_eltype
         ## @test b[Block()[]] == 2
       end
     end
+
+    # adjoint and transpose
+    a = dev(BlockSparseArray{elt}([1], [1, 1]))
+    @allowscalar a[1, 2] = 1
+    @test [@views(a[it]) for it in block_stored_indices(a)] isa Vector
+    ah = adjoint(a)
+    @test [@views(ah[it]) for it in block_stored_indices(ah)] isa Vector
+    at = transpose(a)
+    @test [@views(at[it]) for it in block_stored_indices(at)] isa Vector
```

Comment:
Could you make these tests a bit more comprehensive, i.e. maybe set some blocks and then check they are set correctly, check the types of the blocks, etc.?